pkg/nmi/server/server.go

```diff
@@ -218,18 +219,33 @@ func (s *Server) hostHandler(w http.ResponseWriter, r *http.Request) (ns string)
 		http.Error(w, "request remote address is not from a host", http.StatusInternalServerError)
 		return
 	}
-	if !validateResourceParamExists(rqResource) {
+	if !tokenRequest.ValidateResourceParamExists() {
 		klog.Warning("parameter resource cannot be empty")
 		http.Error(w, "parameter resource cannot be empty", http.StatusBadRequest)
 		return
 	}
-	podID, err := s.TokenClient.GetIdentities(r.Context(), podns, podname, rqClientID)
+
+	if len(tokenRequest.ResourceID) > 0 {
```

Comment:
So the filtering of `AzureIdentity` is done in `standard.go` and `managed.go`.

The user creates an `AzureIdentity`, `AzureIdentityBinding` and then the pods. When the pod gets created, MIC assigns the identity to the underlying node in case of standard mode and then creates an `AzureAssignedIdentity` which maps all 3 objects (azid, azidbinding and the pod). The `AzureAssignedIdentity` already contains the `AzureIdentity` ref which we use to filter [here](https://github.com/Azure/aad-pod-identity/blob/master/pkg/nmi/standard.go#L84-L151) based on clientID being present/not in the request. So we would need to move this logic into standard and managed modes.